integration_test/scenarios/web_socket_update_ui_scenario.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:tmail_ui_user/features/thread/presentation/widgets/email_tile_builder.dart';
+
+import '../base/base_scenario.dart';
+import '../models/provisioning_email.dart';
+import '../utils/scenario_utils_mixin.dart';
+import 'login_with_basic_auth_scenario.dart';
+
+class WebSocketUpdateUiScenario extends BaseScenario with ScenarioUtilsMixin {
+  const WebSocketUpdateUiScenario(super.$, {
+    required this.loginWithBasicAuthScenario,
+    required this.subject,
+    required this.content,
+  });
+
+  final LoginWithBasicAuthScenario loginWithBasicAuthScenario;
+  final String subject;
+  final String content;
+
+  @override
+  Future<void> execute() async {
+    await loginWithBasicAuthScenario.execute();
+
+    await provisionEmail(
+      [ProvisioningEmail(
+        toEmail: loginWithBasicAuthScenario.email,
+        subject: subject,
+        content: content,
+      )],
+      refreshEmailView: false,
+    );
+    await $.pumpAndSettle();
+    await _expectEmailVisible(loginWithBasicAuthScenario.email);
+    await _expectEmailUnreadWithSubject(subject);
+    await _expectEmailUnstarredWithSubject(subject);
+
+    markEmailsAsReadWithSubjects(
+      subjects: [subject],
+      isRead: true,
+    );
+    await $.pumpAndSettle();
+    await _expectEmailReadWithSubject(subject);
```

Comment:
Because right now, we still have `refresh` after each action, so maybe this test will not have much value?

Reply:
Updated